pkg/piecestore/rpc/server/server.go

```diff
@@ -53,24 +56,83 @@ func (c Config) Run(ctx context.Context, server *provider.Provider) (err error)
 	return server.Run(ctx)
 }
 
+//DirSize returns the total size of the files in that directory
+func DirSize(path string) (int64, error) {
+	var size int64
+	_, err := os.Stat(path)
+	if err != nil {
+		return 0, errors.New("path doesn't exists")
+	}
+	adjSize := func(_ string, info os.FileInfo, err error) error {
+		if !info.IsDir() {
+			size += info.Size()
+		}
+		return err
+	}
+	err = filepath.Walk(path, adjSize)
+
+	return size, err
+}
+
 // Server -- GRPC server meta data used in route calls
 type Server struct {
-	DataDir string
-	DB      *psdb.DB
-	pkey    crypto.PrivateKey
+	DataDir        string
+	DB             *psdb.DB
+	pkey           crypto.PrivateKey
+	totalAllocated int64
 }
 
 // Initialize -- initializes a server struct
 func Initialize(ctx context.Context, config Config, pkey crypto.PrivateKey) (*Server, error) {
 	dbPath := filepath.Join(config.Path, "piecestore.db")
 	dataDir := filepath.Join(config.Path, "piece-store-data")
 
+	// read the allocated disk space from the config file
+	allocatedDiskSpace := config.AllocatedDiskSpace
+
+	// get the disk space details
+	diskSpace, err := disk.Usage("/")
+	freeDiskSpace := int64(diskSpace.Free)
+	if err != nil {
+		return nil, err
+	}
+
 	db, err := psdb.Open(ctx, dataDir, dbPath)
 	if err != nil {
 		return nil, err
 	}
 
-	return &Server{DataDir: dataDir, DB: db, pkey: pkey}, nil
+	// get how much is currently used, if for the first time totalUsed = 0
+	totalUsed, err := DirSize(dataDir)
```

Comment:
use the database call instead of the file system check to get used space

Reply:
done